src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -2389,10 +2389,7 @@ void Assembler::ldmxcsr( Address src) {
 
 void Assembler::leal(Register dst, Address src) {
   InstructionMark im(this);
-#ifdef _LP64
-  emit_int8(0x67); // addr32
```

Comment:
Partial writes only happen with data size smaller than 32-bit since 32-bit writes are zero-extended to full register width. As a result, there should be no additional stalls here.

Reply:
> By removing the 0x67 prefix, we are now doing 64 bit computation even for 32 bit inputs. Wouldn't this cause partial register stalls (see section 3.5.2.4 end of section).

I agree with @merykitty .
The manual mentions
```
Note that in Intel 64 architecture, an update to the lower 32 bits of a 64 bit integer register is architecturally defined to zero extend the upper 32 bits. While this action may be logically viewed as a 32 bit
update, it is really a 64 bit update (and therefore does not cause a partial stall).
```

TBH, I didn't know this opt rule before.
Thanks @sviswa7 for sharing this.